### PR TITLE
Improve example server logging.

### PR DESCRIPTION
**Pull request by rillian**
Report initial mds vector as well as how frequently and which md
is punctured by the background task.

### CODE REVIEW

File: ppoprf/examples/server.rs [hunk 1]
```diff
@@ -141,20 +141,28 @@ async fn main() -> std::io::Result<()> {
...
             if let Ok(mut state) = background_state.write() {
-                log::info!("Epoch rotation: puncturing '{:?}'", md);
+                info!("Epoch rotation: puncturing '{:?}'", md);
                 state.prf_server.puncture(md);
                 state.md_idx += 1;
             }
-            log::info!("Epoch rotation complete");
+            if let Ok(state) = background_state.read() {
+                if state.md_idx >= mds.len() {
+                    warn!("All epoch tags punctured! No further evaluations possible.");
```
Comment (DJAndries) [inline ppoprf/examples/server.rs]: what if we placed this warning log right after the loop...

File: ppoprf/examples/server.rs [hunk 2]
```diff
@@ -141,20 +141,28 @@ async fn main() -> std::io::Result<()> {
...
+                } else {
+                    let md = &mds[state.md_idx];
+                    info!("Epoch tag now '{:?}'; next rotation in {} seconds", md, epoch.as_secs());
```
Comment (DJAndries) [inline ppoprf/examples/server.rs]: ... and took this info log out of the if condition?

**Comment by DJAndries** [comment]
> r? @DJAndries

couple comments, but all is good!

File: ppoprf/examples/server.rs (hunk 1 above)
Comment (rillian) [reply ppoprf/examples/server.rs]: That's a little cleaner, thanks.

File: ppoprf/examples/server.rs (hunk 2 above)
Comment (rillian) [reply ppoprf/examples/server.rs]: We still need the conditional. In the last iteration of the loop we increment `md_idx` so it points one past the end of the `mds` vector, so the final log statement will panic with `index out of bounds`.

Comment (DJAndries) [reply ppoprf/examples/server.rs]: i see. would it make sense to take this info log and put it right at the beginning of the loop? There would be no need to index `mds` in that case

Comment (rillian) [reply ppoprf/examples/server.rs]: That's a good idea. Thanks!

**Pull request merged**